release(v1.11.0-rc.0): prepare release (#11612)
siderolabs/talos: CHANGELOG.md

## [Talos 1.11.0-rc.0](https://github.com/siderolabs/talos/releases/tag/v1.11.0-rc.0) (2025-08-18)

Welcome to the v1.11.0-rc.0 release of Talos!  
*This is a pre-release of Talos*

Please try out the release binaries and report any issues at
https://github.com/siderolabs/talos/issues.

### Azure

Talos on Azure now defaults to MTU of 1400 bytes for the `eth0` interface to avoid packet fragmentation issues.

> The default MTU can be overriden with machine configuration.

Reviewer: ```suggestion
The default MTU can be overriden via machine configuration.
```

### Boot

> Talos increases the boot partition size to 2 GiB to accommodate larger images (with many system extensions included).

Reviewer: ```suggestion
Talos boot partition size increased to 2 GiB to accommodate large images (with many system extensions included).
```

### Kernel Command Line

Talos now exposes the kernel command line as a KernelCmdline resource (`talosctl get cmdline`).

### Disk Encryption

Disk encryption for system volumes is now managed by the `VolumeConfig` machine configuration document.
Legacy configuration in `valpha1` machine configuration is still supported.

New per-key option `lockToSTATE` is added to the `VolumeConfig` document, which allows to lock the volume encryption key to the secret salt in the `STATE` volume.
So, if the `STATE` volume is wiped or replaced, the volume encryption key will not be usable anymore.

### Disk Wipe

Talos now supports `talosctl disk wipe` command in maintenance mode (`talosctl disk wipe <disk> --insecure`).

### Early Inline Configuration

Talos now supports passing early inline configuration via the `talos.config.early` kernel parameter.
This allows to pass the configuration before the platform config source is probed, which is useful for early boot configuration.

> The value of this parameter has same format as the `talos.config.inline` parameter, i.e. it should be base64 encoded and zstd-compressed.

Reviewer: ```suggestion
The value of this parameter has the same format as the `talos.config.inline` parameter, i.e. it should be base64 encoded and zstd-compressed.
```

### ETCD downgrade API

Added ETCD downgrade API mimicking the ETCD API and etcdctl interfaces.
This API allows to downgrade ETCD cluster (storage format) to a previous version.

### IMA support removed

Talos now drops the IMA (Integrity Measurement Architecture) support. This feature was not used in Talos for any meaningful security purpose
and has historically caused performance issues. See #11133 for more details.

### Kubernetes Version Validation

> Talos now validates Kubernetes version in the image submitted in the machine configuration.

Reviewer: ```suggestion
Talos now validates the Kubernetes version in the image specified in the machine configuration.
```
